query/physicalplan/sampler.go

```diff
@@ -142,12 +201,70 @@ func (s *ReservoirSampler) Finish(ctx context.Context) error {
 
 // replace will replace the row at index i with the row in the record r at index j.
 func (s *ReservoirSampler) replace(i int, newRow sample) {
-	s.reservoir[i].r.Release()
+	s.sizeInBytes -= s.reservoir[i].Release()
 	s.reservoir[i] = newRow
```

Comment:
Is this valid? Could you replace a sample that references a whole record (i.e. whose index is -1) with a sample that references just a single row?

Reply:
No, because we always slice the reservoir on fill. So we can't reach replace code with records that have are unsampled (-1)